src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -1011,4 +1011,18 @@ private boolean isVariableMatchesIgnorePattern(String variable) {
         return matcher.matches();
     }
 
+    /**
+     * Check if the statement is variable definition or resource or modifier or type.
+     *
+     * @param type
+     *        Type of the statement.
+     * @return true if the statement is, otherwise false.
+     */
+    private static boolean isStatementVariableDefinition(int type) {
+        return type == TokenTypes.VARIABLE_DEF
+                || type == TokenTypes.TYPE
+                || type == TokenTypes.MODIFIERS
+                || type == TokenTypes.RESOURCE;
+    }
+
```

Comment:
This method is confusing, `isStatementVariableDefinition`, I think the correct way of checking whether any AST token is part of the variable definition is by checking if one of its ancestor ast is `VARIABLE_DEF`.

If this method does not aim to do that, the name should be changed or split into different methods if a single method can not have a proper name.

Reply:
> This method is confusing, `isStatementVariableDefinition`, I think the correct way of checking whether any AST token is part of the variable definition is by checking if one of its ancestor ast is `VARIABLE_DEF`.
> 
> If this method does not aim to do that, the name should be changed or split into different methods if a single method can not have a proper name.

Okay. Done. I have changed the name of the function.

Reply:
Because I think the code will be easier to understand with conditions combined into a function.

Reply:
But the function can't be named appropriately, combining conditions that do a lot of things without appropriate naming is not worth it.

Reply:
Romani suggested a name as isZeroDistanceToken. It makes more sense than isDefinition.